### ISSUE TITLE
Exporting Delivery Groups

### ISSUE THREAD

**Issue by Spontrel** [ryancbutler/XDReplicate#27]
## Expected Behavior
 
Trying to export a single DG
 
N/A

## Current Behavior

Doesn't accept command: Export-XDSite  -Verbose -xdhost localhost -XMLPATH "\\mydomain\XA7_replication\Module_Replication.xml" -dgtag Test

** Does work for all DGs if nothing is selected.  Also, if -ignoredgtag is used for one of my two DGs, it allows the script to run, but it still processes both of my two DGs.

error

VERBOSE: BEGIN: Export-XDsite
VERBOSE: Checking for Citrix Snapins
VERBOSE: Found snapins
VERBOSE: END: Test-XDmodule
NO DELIVERY GROUPS FOUND
At C:\Windows\system32\WindowsPowerShell\v1.0\Modules\CTXXD-Replicate\Public\Ex
port-XDsite.ps1:71 char:9
+         throw "NO DELIVERY GROUPS FOUND"
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (NO DELIVERY GROUPS FOUND:Stri 
   ng) [], RuntimeException
    + FullyQualifiedErrorId : NO DELIVERY GROUPS FOUND

## Possible Solution

Dunno yet...still learning your new code

## Steps to Reproduce (for bugs)

1.  load module

2. Export-XDSite  -Verbose -xdhost localhost -XMLPATH "\\mydomain\XA7_replication\Module_Replication.xml" -dgtag Test
3. get error.
4.

## Your Environment

* Module Version used:
* Citrix XenDesktop Version:

Export-XDSite  -Verbose -xdhost localhost -XMLPATH "\\amc\orca\scripts\citrix_XA7\XA7_replication\Module_Replication.xml" -dgtag Test

**Comment by ryancbutler** [comment]
The value for dgtag or ignoredgtag is the actual "tag" name attached to the delivery group.  Not the delivery group name itself.  Maybe I can explain this better somehow?  For example from the image below I would use `-dgtag "group1"` to include and `-ignoredgtag "group1"` to exclude from the export.
![image](https://user-images.githubusercontent.com/14218733/30997145-731453ca-a48b-11e7-9012-470f574146b0.png)

**Comment by Spontrel** [comment]
Yes, you are correct, I totally blew it on that one.  I renamed the Tag on my Delivery Group so it stood out from other tags - oops.  That said, I fixed my command line and the export now works error free.  That was totally my fault.  I'm sorry.

I did run the export successfully.  I chose the correct Delivery Group tag for one of my two Delivery Groups and watched it in verbose mode.  The export processes all apps both of the delivery groups, as if it doesn't honor the tag at all.  Also, on the import, there are the following errors:

BEGIN: Test-XDBrokerAdminFolder
VERBOSE: Processing Folder Replicated\Troubleshooting\Apps\
VERBOSE: FOLDER FOUND
VERBOSE: Multiple DG
VERBOSE: BEGIN: New-XDappobject
VERBOSE: BEGIN: Set-XDAppUserPerm
VERBOSE: Setting App Permissions
VERBOSE: domain\Testers
VERBOSE: domain\Admins
VERBOSE: END: Set-XDAppUserPerm
VERBOSE: Configuring App for multiple DG and AG
VERBOSE: BEGIN: Set-XDMultiApp
VERBOSE: Setting Replicated\Troubleshooting\Apps\TEST22
VERBOSE: Setting Application Group(s) for TEST22
ADD-BrokerApplication : No items match the supplied pattern
At C:\Windows\system32\WindowsPowerShell\v1.0\Modules\CTXXD-Replicate\Private\S
et-XDMultiApp.ps1:96 char:17
+                 ADD-BrokerApplication -InputObject $appmatch 
-ApplicationGroup $ ...
+                 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Add-BrokerApplication], Sd 
   kOperationException
    + FullyQualifiedErrorId : Citrix.XDPowerShell.Broker.NoMatchingItems,Citri 
   x.Broker.Admin.SDK.AddBrokerApplicationCommand
 
VERBOSE: END: Set-XDMultiApp

**Comment by Spontrel** [comment]
I also tested with -ignoreapptag choosing the DG tag I do not want, and I'm not seeing errors in the import.  Although there are no errors, I am getting apps built in the DG tag that I wanted to ignore...so it is not honoring the ignore in the export.

**Comment by ryancbutler** [comment]
The app tag is associated to the actual application not the DG.  The DG tag was originally ignoring\adding applications associated to the DG to the export but the problem is now applications can be associated to multiple DGs which makes it more difficult to code.  As of now the the DG tag only works with availability groups and delivery groups and doesn't touch apps.  And app tags only work for apps.  Need to work through this better now.

I can add the 2000 max in a future version.

**Comment by Spontrel** [comment]
I was able to successfully test using -apptag so that works fine. I rely on using a small test set of apps to speed up testing cycles and not get all of my apps every time. 

With -dgtag I can see the need to scan through all apps while building the export, however it is still making apps in the target site for all Delivery Groups which basically makes it treat the export like I did not make any DGtag. I hope that clarification helps.
